airflow/www/security.py

```diff
@@ -155,6 +154,11 @@ class AirflowSecurityManager(SecurityManager, LoggingMixin):
     ###########################################################################
 
     ROLE_CONFIGS = [
+        {
+            'role': 'Public',
+            'perms': {'can_index'},
```

Comment:
another way to do is to move the public role here and keep the perms / vms empty, but override the perms / vm in custom security manager.

cc @astahlman @ashb WDYT?